# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsInputStreamPump.cpp
# Commit: 74f99033251c
# Full Hash: 74f99033251cd6c36eb6b0b5a4713eb6b5c793fa
# Author: Jonathan Kingston <jkt@mozilla.com>
# Date: 2019-02-13 09:48:26
# Regressor Bug: 1520868
# File Overlap Count: 1
# Description:
#   Bug 1520868 - Replacing AsyncOpen2 with AsyncOpen always r=valentin
#   
#   Replacing js and text occurences of asyncOpen2
#   Replacing open2 with open
#   
# ==============================================================================

diff -r 3378fd5f4f22 -r 74f99033251c netwerk/base/nsInputStreamPump.cpp
--- a/netwerk/base/nsInputStreamPump.cpp	Tue Feb 12 17:58:19 2019 +0100
+++ b/netwerk/base/nsInputStreamPump.cpp	Tue Feb 12 16:08:25 2019 +0000
@@ -353,7 +353,6 @@
 
   mState = STATE_START;
   mListener = listener;
-  mListenerContext = ctxt;
   return NS_OK;
 }
 
@@ -485,7 +484,7 @@
     // deadlocks when calls to RetargetDeliveryTo for multiple
     // nsInputStreamPumps are needed (e.g. nsHttpChannel).
     RecursiveMutexAutoUnlock unlock(mMutex);
-    rv = mListener->OnStartRequest(this, mListenerContext);
+    rv = mListener->OnStartRequest(this, nullptr);
   }
 
   // an error returned from OnStartRequest should cause us to abort; however,
@@ -552,7 +551,7 @@
       // deadlocks when calls to RetargetDeliveryTo for multiple
       // nsInputStreamPumps are needed (e.g. nsHttpChannel).
       RecursiveMutexAutoUnlock unlock(mMutex);
-      rv = mListener->OnDataAvailable(this, mListenerContext, mAsyncStream,
+      rv = mListener->OnDataAvailable(this, nullptr, mAsyncStream,
                                       mStreamOffset, odaAvail);
     }
 
@@ -653,10 +652,9 @@
     // deadlocks when calls to RetargetDeliveryTo for multiple
     // nsInputStreamPumps are needed (e.g. nsHttpChannel).
     RecursiveMutexAutoUnlock unlock(mMutex);
-    mListener->OnStopRequest(this, mListenerContext, mStatus);
+    mListener->OnStopRequest(this, nullptr, mStatus);
   }
   mListener = nullptr;
-  mListenerContext = nullptr;
 
   if (mLoadGroup) mLoadGroup->RemoveRequest(this, nullptr, mStatus);
 